src/Take.Blip.Builder/StateManager.cs

```diff
@@ -22,6 +22,11 @@ public Task<string> GetPreviousStateIdAsync(IContext context, CancellationToken
             return context.GetContextVariableAsync(GetPreviousStateKey(context.Flow.Id), cancellationToken);
         }
 
+        public Task<string> GetParentStateIdAsync(IContext context, CancellationToken cancellationToken)
+        {
+            return context.GetContextVariableAsync(GetStateKey(context.Flow.Parent.Id), cancellationToken);
```

Comment:
I saw somewhere else checking if `context.Flow.Parent` is null. Could it also be null here too?

Reply:
added validation